### PR TITLE
Added conditional display of resources based on device capability. Added new resources: Qtree, Filesystem and Shares for NAS devices.

### LINKED ISSUE #530: [Delfin UI] Add new resource types for only NAS storage device to support NAS capabilities: Qtree, File Systems, File share
**Issue/Feature Description:**
Delfin is adding support for storage devices with NAS capabilities. 
The dashboard should be able to display the following resources for NAS capable devices:

- [ ] Qtree
- [ ] File System
- [ ] File Share

**Why this issue to fixed / feature is needed(give scenarios or use cases):**

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind new feature

**What this PR does / why we need it**:
This PR adds support for conditionally displaying the resources supported by a device. SODA Delfin supports SAN, NAS and  devices with Unified capabilities. Some devices may not have all the resources supported and the Dashboard should not display the resources for such devices.
Currently Delfin supports: Storage Pools, Volumes, Controllers, Ports and Disks display. These are displayed for all devices in the tree view in the storage dashboard and tab view in storage details page.
With this PR we display the resource type only if the device model supports the capability. This capability is maintained as a map at the dashboard in the following format:
```
 resources:{
            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            pools : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            controllers : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            ports : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc'],
            disks : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            qtrees : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            filesystems : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            shares: ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc']
        },
```
In the above Ports and Shares resources are not supported in the fake_driver devices.

The map above has to be updated with the actual support matrix based on the device capability.

This PR also adds new resource types Qtree, Filesystem and Shares to devices that support NAS capabilities.

**Which issue(s) this PR fixes**:
Fixes #530 

**Test Report Added?**:
/kind TESTED

**Test Report**:
nas-8
nas-9
nas-10
nas-1
nas-2
nas-3
nas-4
nas-5
nas-6
nas-7

**Special notes for your reviewer**:
The device capability matrix pushed in this PR enables all resource types for all device models.
The actual support matrix must be updated before the PR is merged or in a subsequent PR.

### CODE REVIEW

**Comment by joseph-v** [comment]
Can we reorder the resource tabs to [Storage pool > Volume > Controller > Port ? Filesystem > Qtree > Share]

**Comment by anvithks** [comment]
> Can we reorder the resource tabs to [Storage pool > Volume > Controller > Port ? Filesystem > Qtree > Share]

@joseph-v This is done. Please review the PR.

**Comment by anvithks** [comment]
@NajmudheenCT @joseph-v @kumarashit 
I have updated this PR and added screenshots.
The resources are now conditionally displayed. 
The resources supported are maintained in the constants file in an object array as mentioned above in the PR description.
We have to update this with actual values. Please help me with the real values to update the PR.

File: src/app/shared/utils/consts.ts [hunk 1]
```diff
@@ -171,6 +174,16 @@ export const Consts = {
                 value: 'ibm'
             }
         ],
+        resources:{
+            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
```
Comment (NajmudheenCT) [inline src/app/shared/utils/consts.ts:178]: resource list need to be based on the current backend support.

current backend support is 
resources:{
            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            pools : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            controllers : ['oceanstor', 'fake_driver']
            ports : ['oceanstor', 'fake_driver']
            disks : ['oceanstor', 'fake_driver'],
            qtrees : ['oceanstor', 'fake_driver'],
            filesystems : ['oceanstor', 'fake_driver'],
            shares: ['oceanstor', 'fake_driver']
        },

File: src/app/shared/utils/consts.ts [hunk 2]
```diff
@@ -171,6 +174,16 @@ export const Consts = {
...
+            pools : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            controllers : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            ports : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc'],
+            disks : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            qtrees : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            filesystems : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            shares: ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc']
```
Comment (NajmudheenCT) [inline src/app/shared/utils/consts.ts:185]: Quotas also we need to add.. for displaying qouata we can either put it as different tab, or is it possible to associate with  filestem and qtree objects?

Comment (anvithks) [reply src/app/shared/utils/consts.ts:185]: I do not have the details for adding Quotas. We can probably take it up as a subsequent task.

File: src/app/shared/utils/consts.ts (hunk 1 above)
Comment (anvithks) [reply src/app/shared/utils/consts.ts]: Thanks @NajmudheenCT I have updated the PR with this. 
I removed `fake_driver` as it is only needed for development.

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Review by joseph-v — APPROVED** [review]
LGTM

File: src/app/shared/utils/consts.ts (hunk 2 above)
Comment (anvithks) [reply src/app/shared/utils/consts.ts]: @NajmudheenCT This has been logged as an issue #539

**Review by kumarashit — APPROVED** [review]
LGTM

**Pull request merged**